### PR TITLE
Add a note to guard against a concern of Michael's for the ZMQ module

### PR DESCRIPTION
**Pull request by lydia-duncan**
Michael wants to be certain that if we fail to provide support for a particular
ZMQ option, the user has a way to work around it until we do support it.  With
setsockopt going away, the current strategy relies on the Socket classRef field
not being private.  Leave a note on that field so we don't forget about that
benefit when the day comes where we can make it private.

### CODE REVIEW

File: modules/packages/ZMQ.chpl [hunk 1]
```diff
@@ -609,6 +609,10 @@ module ZMQ {
     Note that this record contains private fields not listed below.
    */
   record Socket {
+    // Note: if we make this private but haven't exposed all the setsockopt
+    // options, users will need another way to work around that lack of support.
+    // Currently, they can work around it by defining their own extern version
+    // and using this field.
```
Comment (mppf) [inline modules/packages/ZMQ.chpl]: I would think that if this is part of certain use cases, we'd want to actually document this field

**Comment by lydia-duncan** [comment]
> perhaps an issue describing that step would be better?

I'm concerned that private fields are far enough in the future that we will forget to look for the ZMQ "what to do when we make classRef private" issue without a note on it.  I'm okay with making both, but I don't want to replace the comment with just an issue

**Comment by mppf** [comment]
That's fine, then. Anyway I'd appreciate the issue & the comment could include the issue number. Thanks!

**Comment by lydia-duncan** [comment]
Created #13503, thanks for the feedback!

**Pull request merged**